### PR TITLE
Conditional export

### PR DESCRIPTION
**Pull request by pburnsdata**
# Issues
#45 
#41
#48
#47

### CODE REVIEW

**Comment by pburnsdata** [comment]
Right now I am adding subject level info if the subject is the lowest nest, and removing total rows only for this subject level export - wasn't sure if totals might be useful for general exports. Also, there is a bug here where you can't do one export type then another and still get the subject level info (e.g. click csv then click export when subject is lowest nest) it's driving me nuts, but maybe it'll be obvious with some fresh eyes. this.data.filtered seems to be updated during deriveData.js for some reason... I've tried deleting the export and reset the filtered data etc. but the subject level info always seems to stay after its first use and atleast that is causing problems when going form csv to xlsx export.

File: src/configuration/rendererSettings.js [hunk 1]
```diff
@@ -1,6 +1,27 @@
 export default function rendererSettings() {
     return {
-        id_cols: ['sitename', 'subjectnameoridentifier'],
+        nestings: [
```
Comment (samussiah) [inline src/configuration/rendererSettings.js]: very nice

File: src/configuration/rendererSettings.js [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+            },
+            {
+                value_col: 'foldername',
+                label: 'Folder',
+                default: false
+            },
+            {
+                value_col: 'formoid',
+                label: 'Form',
+                default: false
```
Comment (samussiah) [inline src/configuration/rendererSettings.js]: what if all four default to `true`?

File: src/configuration/syncSettings.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export default function syncSettings(settings) {
+    settings.id_cols = settings.nestings.filter(d => d.default === true).map(f => f.value_col);
```
Comment (samussiah) [inline src/configuration/syncSettings.js]: Sweet array!  In the edge case that all four nestings have `d.default === true` maybe `.slice(0,3)` after the `.filter()`.

File: src/configuration/rendererSettings.js (hunk 2 above)
Comment (pburnsdata) [reply src/configuration/rendererSettings.js]: a sobering question haha, ill probably add what you commented below

**Pull request merged**